app/routes.py

```diff
@@ -149,26 +161,37 @@ def generator():
 
 
 async def generator(service_object, request):
-    total_downloaded = 0
-    partial_downloaded = 0
+    progress_mapping = {
+        Status.DOWNLOADING: lambda evt: evt["progressDetail"].get("current", 0)
+        / evt["progressDetail"].get("total", 1),
+        Status.PULLING_FS_LAYER: lambda _: 0,
+        Status.WAITING: lambda _: 0,
+        Status.VERIFYING_CHECKSUM: lambda _: 97,
+        Status.DOWNLOAD_COMPLETE: lambda _: 98,
+        Status.EXTRACTING: lambda _: 99,
+        Status.PULL_COMPLETE: lambda _: 100,
+    }
+    layers = {}
 
     client = utils.get_docker_client()
+
```

Comment:
Please handle exception(with try/except) when invoking docker.

Reply:
this is the original code, this is how docker is getting invoked in code base